BLSQ/iaso
# Polio 1358 thousand separator

We needed number formatting for vaccine supplychain, it impacts all of iaso

Related JIRA tickets : POLIO-1358

## Self proofreading checklist


## Changes

- Import `NumberInput` and `FormattedNumber` from `bluesquare-components`
- Refactor `InputComponent` to use `NumberInput`. Add props and get thousand and decimal separators from the locale by default
- Add `NumberCell` which uses `FormattedNumber` with the same defaults as `InputComponent` but can be used in tables and text.
- Update all components in polio that used `TextInput` to show numbers so they use `NumberInput` instead
- Update (hopefully) all displayed number in iaso to display formatted numbers. ids, and submission fields were left as they were


## How to test
- run `npm ci`
- replace dep to bluesquare-components in package.json :
`"bluesquare-components": "github:BLSQ/bluesquare-components#POLIO-1358_thousand_separator",`
- run `npm update bluesquare-components && npm run dev` (you will run webpack outside of docker)
- Go through iaso and check that numbers are formatted where needed. 
- Also go through polio

What you should see: when setting language to english, thousands should be separated by commas, and decimals by a dot. When setting language to French it should be the opposite.

So when checking high integers, check in english, when checkinf floats (like latitude/longitude), check in french.

I tried to make 1 commit per domain, so reviewers can go through the commit list to have an idea of the domains to test/double check

**Note**: play special attention to entities, planning, DHIS2 mappings and completeness as I didn't have a ton of local data to test these features

## Print screen / video

There's too much really.

## Notes
⚠️Depends on https://github.com/BLSQ/bluesquare-components/pull/137

TESTS WILL FAIL UNTIL https://github.com/BLSQ/bluesquare-components/pull/137 is merged and the dep updated
